indico/modules/events/registration/fields/simple.py

```diff
@@ -472,7 +472,7 @@ def _stay_dates_valid(form, field):
                 raise ValidationError(_("Arrival date can't be set after the departure date."))
 
         def _check_number_of_places(form, field):
-            if not field.data:
+            if not field.data or not field.data.get('choice'):
```

Comment:
I still think this is not needed anymore now that you changed it to only save field data if an accommodation has been selected. (unless you still have data in your DB with `choice: ""`)